app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,29 @@
+class SessionsController < ApplicationController
+  def front
+    redirect_to videos_path if sign_in?
+  end
+
+  def new
+    redirect_to videos_path if sign_in?
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect_to videos_path
+    else
+      flash[:danger] = "Incorrect email or password, Please try again"
```

Comment:
Flash[:notice]’s message will persist to the next action and should be used when redirecting to another action via the ‘redirect_to’ method.

Flash.now[:notice]’s message will be displayed in the view your are rendering via the ‘render’ method.